src/facilities-context/FacilitiesContext.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect } from "react";
+
+import { Facility } from "../page-multi-facility/types";
+import useScenario from "../scenario-context/useScenario";
+import { facilitiesReducer } from "./reducer";
+import * as actions from "./actions";
+
+export type FacilityMapping = { [key in Facility["id"]]: Facility }
+
+export interface FacilitiesState {
+  loading: boolean;
+  failed: boolean;
+  facilities: FacilityMapping;
+}
+
+export type FacilitiesDispatch = (action: actions.FacilitiesActions) => void;
+
+export type ExportedAsyncActions = {
+  updateFacility: (...props: any) => Promise<void>;
+  removeFacility: (...props: any) => Promise<void>;
+  // fetchRtData: ExportedAsyncFunction;
+}
+
+export interface FacilitiesContext {
+  state: FacilitiesState;
+  dispatch: FacilitiesDispatch | undefined;
+  asyncActions: ExportedAsyncActions;
+}
+
+export const FacilitiesContext = React.createContext<FacilitiesContext | undefined>(
+  undefined,
+);
+
+export const FacilitiesProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [state, dispatch] = React.useReducer(facilitiesReducer, {
+    loading: true,
+    failed: false,
+    facilities: {},
+  });
+
+  const [scenario] = useScenario();
+  const scenarioId = scenario?.data?.id
```

Comment:
I'm not entirely sure that this will work without issues, I haven't seen examples of a context hook being used like this. I would assume it would work because we are nesting `FacilitiesProvider` after `ScenarioProvider`, but would need to do some testing to ensure that switching scenarios updates this context.

Reply:
Ohh cool, I will try that